plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -221,9 +213,14 @@ func (p *Postgresql) accRow(meas_name string, row scanner, acc telegraf.Accumula
 		return err
 	}
 
-	if columnMap["datname"] != nil {
+	if c, ok := columnMap["datname"]; ok && *c != nil {
 		// extract the database name from the column map
-		dbname.WriteString((*columnMap["datname"]).(string))
+		switch (*c).(type) {
```

Comment:
Use a variable `switch datname := (*c).(type) {` so you don't need to redo the conversion in the case.